### ISSUE TITLE
foreign_keys and exit options do not overwrite color as noted in README

### ISSUE THREAD

**Issue by mintelm** [nvimtools/hydra.nvim#37]
Hello everyone,

there's a sentence in the README that describes that `foreign_keys` and `exit` configuration options actually overwrite the behavior specified by a color. At least I could not get it working.

What is my problem?
I want to create a DAP hydra that has the policy of an `amaranth` hydra. However, since DAP stepping/continuing is asynchronous, I have to use a pink hydra. When using a pink hydra and specifying `{ foreign_keys = 'warn', exit = true }`, the behavior of the hydra does not change, i.e. it still processes inputs that are not mapped to a head.

What would solve my problem?
- option 1: I can create a pink hydra and specify `foreign_keys` and `exit` and the configuration actually changes the behavior of the hydra
- option 2: I can create a amaranth hydra that works with the asynchronous events of DAP (I tried the trick with `{ on_key = function() vim.wait(20) end, color = 'amaranth' }` that is used in the proposed git hydra, however that does not work as expected with the DAP since when i use `dap.continue()` it actually pastes some text into the popup created by `dap.continue()`

Example code:
```
    hydra({
        name = 'Debugging',
        config = { color = 'pink', foreign_keys = 'warn', exit = true },
        hint = hints.dap,
        mode = 'n',
        body = '<Leader>d',
        heads = {
            { 'b',     cmd('DapToggleBreakpoint') },
            { 'c',     cmd('DapContinue') },
            { 's',     cmd('DapStepOver') },
            { 'i',     cmd('DapStepInto') },
            { 'o',     cmd('DapStepOut') },
            { 'r',     cmd('DapToggleRepl') },
            { 'u',     function() require('dapui').toggle({ reset = true }) end },
            { 'w',     function() require('dapui').elements.watches.add() end },
            { '<Esc>', nil,                                                     { exit = true, nowait = true, desc = false }, },
        },
    })

```

**Comment by benlubas** [comment]
https://github.com/benlubas/hydra.nvim/blob/eb68396f4f3185e000d64711419c71f67b23bb1d/lua/hydra/init.lua#L126

It seems like there's an inconsistency between the docs and the code here.

**Comment by benlubas** [comment]
going back to when that code was written, the docs said the same thing. Guess it just slipped through at the time and hasn't been caught since.

However, I'm not sure fixing this will even fix the problem you're having. `color` and `exit`/`foreign_keys` do have to be reconciled one way or the other, so you currently can't have a Pink hydra that behaves like an Amaranth one, that would just be made into an Amaranth hydra.

**Comment by mintelm** [comment]
ahh okay, makes sense that it does not work how I expect it to if the documentation is inconsistent.

Okay, if I understood you correctly: with the current implementation it is basically not possible to have an -- I'll use a different terminology here -- _async_ hydra (pink) that uses `foreign_keys = 'warn'`?

**Comment by benlubas** [comment]
Yeah currently impossible. I'd be open to changing it though (at the very least I will update the docs). I'm not sure how hard it would be to do though.

**Comment by mintelm** [comment]
I mean sure, I am keen on this feature, so I'd be glad if you tried it.

**Comment by sufftea** [comment]
I think I'm missing something. The docs don't seem to say anything about asynchrony. In my understanding, there should be *no difference at all* between a pink hydra with { foreign_keys = 'warn', exit = true } and an amaranth hydra.

What do colors have to do with asynchrony?

I really like this plugin as a concept, and am trying to make it work for me, but I've encountered a lot of bugs that make this really hard. 

For example, `red` heads only trigger when i exit hyrda. That is, with the config below, to print "hmm A", I would have to pres `<leader>'` to enter hydra, then `a` (but the command won't be executed yet), and then `esc` -- and only then will "hmm A" be printed. Changing color to `pink` fixes that.
```lua
require("hydra")({
        name = "[HYDRA TEST]",
        -- hint = hint,
        config = {
          color = "pink",
          invoke_on_body = true,
          foreign_keys = "run",
          hint = {
            type = "statusline",
          },
        },
        mode = { "n" },
        body = "<leader>'",
        heads = {
          { "a", "<cmd>echo 'hmm A'<CR>", { private = true, desc = "previous" } },
          { "b", "<cmd>echo 'hmm B'<CR>", { private = true, desc = "next" } },
          { "c", "<cmd>echo 'hmm C'<CR>", { private = true, desc = "delete" } },
          { "d", "<cmd>echo 'hmm D'<CR>", { private = true, desc = "tabnew" } },
          { "<Esc>", nil, { private = true, exit = true, nowait = true, desc = "exit" } },
        },
      })
```

I don't see any feature described in the readme that would explain this behavior. But after seeing you talk about async colors, I'm starting to think those are not actually bugs, and I'm just missing a huge chunk of information about this plugin...
